nav2_map_server/src/occ_grid_loader.cpp

```diff
@@ -126,24 +127,31 @@ OccGridLoader::LoadParameters OccGridLoader::load_map_yaml(const std::string & y
   return loadParameters;
 }
 
-nav2_util::CallbackReturn OccGridLoader::on_configure(const rclcpp_lifecycle::State & /*state*/)
+bool OccGridLoader::loadMapFromYaml(
+  std::string yaml_file,
+  std::shared_ptr<nav2_msgs::srv::LoadMap::Response> response)
 {
-  RCLCPP_INFO(node_->get_logger(), "OccGridLoader: Configuring");
-
-  msg_ = std::make_unique<nav_msgs::msg::OccupancyGrid>();
+  if (yaml_file == "") {
+    RCLCPP_ERROR(node_->get_logger(), "YAML file name is empty, can't load!");
+    response->result = nav2_msgs::srv::LoadMap::Response::RESULT_MAP_ID_DOES_NOT_EXIST;
+    return false;
+  }
+  RCLCPP_INFO(node_->get_logger(), "Loading yaml file: %s", yaml_file.c_str());
   LoadParameters loadParameters;
   try {
-    loadParameters = load_map_yaml(yaml_filename_);
+    loadParameters = load_map_yaml(yaml_file);
   } catch (YAML::Exception & e) {
     RCLCPP_ERROR(
       node_->get_logger(), "Failed processing YAML file %s at position (%d:%d) for reason: %s",
-      yaml_filename_.c_str(), e.mark.line, e.mark.column, e.what());
-    throw std::runtime_error("Failed to load map yaml file.");
+      yaml_file.c_str(), e.mark.line, e.mark.column, e.what());
+    response->result = nav2_msgs::srv::LoadMap::Response::RESULT_INVALID_MAP_METADATA;
```

Comment:
For the error codes, the `---` separation in the message format actually namespaces as you see, so you dont need the `RESULT_` prefix on any of the consts

Reply:
RESULT is the field name (response->result) in this case. I thought I'm following convention here but will change it if I'm not.

Reply:
To be clear, I'm suggesting changing `nav2_msgs::srv::LoadMap::Response::RESULT_INVALID_MAP_METADATA` to `nav2_msgs::srv::LoadMap::Response::INVALID_MAP_METADATA` because the `RESULT_` is implied by the `nav2_msgs::srv::LoadMap::Response::` namespace.

Reply:
So I started with this one: https://github.com/ros/common_msgs/pull/95/files#diff-98c6ab129d291926bae2ab2d4e920e88

It had RESULT_ for all the results, which I think clarifies which field (`result`) in the response the values  are used for. Similarly I used the TYPE_ in the request portion of the .srv file to clarify those are for the `type` field in the request.

I'm not able to find a convention on this however, so I think either one is OK. I think the way I have it is a little more clear, albeit more wordy.